workflow-testing/src/test/java/com/squareup/workflow1/WorkflowsLifecycleTests.kt

```diff
@@ -143,19 +156,20 @@ class WorkflowsLifecycleTests {
 
         val (_, setState) = awaitNextRendering()
         // 2 actions queued up - should start the side effect and then stop it
-        // on two consecutive render passes.
+        // on two consecutive render passes, never actually running the side effect.
         setState.invoke(1)
         setState.invoke(2)
+        dispatcher.scheduler.advanceUntilIdle()
         dispatcher.scheduler.runCurrent()
-        awaitNextRendering()
-        dispatcher.scheduler.runCurrent()
-        if (!runtimeConfig.contains(CONFLATE_STALE_RENDERINGS)) {
+        if (!runtimeConfig.contains(CONFLATE_STALE_RENDERINGS) &&
+          !runtimeConfig.contains(DRAIN_EXCLUSIVE_ACTIONS)
+        ) {
           // 2 rendering or 1 depending on runtime config.
           awaitNextRendering()
         }
 
-        assertEquals(1, started, "Side Effect not started 1 time.")
-        assertEquals(1, cancelled, "Side Effect not cancelled 1 time.")
+        assertEquals(0, started, "Side Effect not started.")
+        assertEquals(0, cancelled, "Side Effect not cancelled.")
```

Comment:
This test was just wrong before. The behavior it is supposed to test is the current behavior of the runtime where if we start and stop rendering a workflow in the same main thread message its side effect never gets executed, because a LAZY start means it needs a dispatch to happen.

We were avoiding this before by using just `runCurrent()` rather than `advanceUntilIdle()` after causing the 2 different actions to be invoked.

Now we are actually testing what we are saying we intended to test.